### ISSUE TITLE
Don't warn on `into_*` functions that take `Self` as parameter

### ISSUE THREAD

**Issue by torkleyy** [intellij-rust/intellij-rust#2416]
I have the following associated function

```rust
    pub fn into_raw(this: Self) -> *mut RawNodeData {
        let raw = this.data.raw();

        forget(this);

        raw
    }
```

This is a pretty common pattern, so I think the warning should not trigger if the only parameter is of type `Self`.
